### PR TITLE
Add target check to PR template

### PR DESCRIPTION
**Pull request by KrishnaIyer**
#### Summary

Add a target check to PR template.

#### Changes

- Ensure that PR authors check the target branch and tick the `Target` item only if the condition is satisfied.

#### Notes for Reviewers

I've pinged the entire stack here for review here to make sure everyone is aware of this and will follow it in their PRs.

#### Checklist

- [ ] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, database and configuration, according to the stability commitments in `README.md`.
- [ ] Testing: The changes are covered with unit tests. The changes are tested manually as well.
- [ ] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -38,3 +38,4 @@ implementation proposal in the referenced issues.
 - [ ] Documentation: Relevant documentation is added or updated.
 - [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
 - [ ] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.
+- [ ] Target: The changes do not contain API updates or require a DB Migration and is hence targeting `master`.
```
Comment (johanstokking) [inline .github/PULL_REQUEST_TEMPLATE.md:41]: I would reword this to:

Target: If the changes are not changing any API, configuration and storage, the target branch is set to `master`, otherwise the target branch is set to `develop`.

Comment (KrishnaIyer) [reply .github/PULL_REQUEST_TEMPLATE.md]: done

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -38,3 +38,4 @@ implementation proposal in the referenced issues.
 - [ ] Documentation: Relevant documentation is added or updated.
 - [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
 - [ ] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.
+- [ ] Target: If the changes are not changing any API, configuration and storage, the target branch is set to master, otherwise the target branch is set to develop.
```
Comment (htdvisser) [inline .github/PULL_REQUEST_TEMPLATE.md:41]: I think there's a bit too much overlap with the "Compatibility" checkbox. Maybe we should combine the two?

Comment (KrishnaIyer) [reply .github/PULL_REQUEST_TEMPLATE.md:41]: Yeah good one. I've combined them

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -33,7 +33,7 @@ implementation proposal in the referenced issues.
 <!-- Make sure that this pull request is complete. -->
 
 - [ ] Scope: The referenced issue is addressed, there are no unrelated changes.
-- [ ] Compatibility: The changes are backwards compatible with existing API, database and configuration, according to the stability commitments in `README.md`.
+- [ ] Compatibility: The changes are backwards compatible with existing API, database and configuration, according to the stability commitments in `README.md`. If yes, the target branch is set to `master`, otherwise the target branch is set to `develop`.
```
Comment (htdvisser) [inline .github/PULL_REQUEST_TEMPLATE.md:36]: This implies that non-compatible changes may go to `develop`, which is not the case. Patches go to `master`. New features, API additions and DB changes go to `develop`. Breaking changes are still not allowed.

Comment (johanstokking) [inline .github/PULL_REQUEST_TEMPLATE.md:36]: This implies that "if no", we can merge anything to `develop` while we also have our compatibility commitment there, although to a lesser extent.

I would reword this to:

```
- [ ] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md`. The target branch is set to `master` if the changes are fully compatible with existing API, database, configuration and CLI.
```

Comment (KrishnaIyer) [reply .github/PULL_REQUEST_TEMPLATE.md]: Please check the updated file.
